.github/workflows/copy_site_to_sourceforge.yml

```diff
@@ -0,0 +1,35 @@
+#############################################################################
+# GitHub Action to Copy Site to SourceForge.
+#
+#############################################################################
+name: "R: Copy Site to SourceForge"
+on:
+  workflow_dispatch:
+    inputs:
+      previous-version:
+        description: 'Previous Version without (-SNAPSHOT)'
```

Comment:
Can we get to know previous version in shell script ?
To not bother maintainer to know what was before.
All others actions need only release version.

Reply:
You could check by tag:
```

➜  checkstyle git:(master) ✗ git tag --sort=creatordate | tail -1
checkstyle-10.4

```

Reply:
How do you see us taking it from folder we are going to archive?

```
[stoyank7@shell-22002 stoyank-checkstyle]$ pwd
/home/project-web/stoyank-checkstyle
[stoyank7@shell-22002 stoyank-checkstyle]$ ls
cgi-bin  checkstyle.github.io.tar.gz  htdocs  htdocs-archive  htdocs-version  old-htdocs-reports  reports  tmp
[stoyank7@shell-22002 stoyank-checkstyle]$ cd htdocs
[stoyank7@shell-22002 htdocs]$ ls
anttask.html                config_naming.html              images                         reports
apidocs                     config_regexp.html              index.html                     running.html
beginning_development.html  config_sizes.html               issue-management.html          scm.html
checks.html                 config_system_properties.html   jdepend-report.html            sponsoring.html
ci-management.html          config_whitespace.html          js                             style_configs.html
cmdline.html                consulting.html                 licenses.html                  styleguides
config.html                 contributing.html               mailing-lists.html             summary.html
config_annotation.html      css                             netbeans.html                  sun_style.html
config_blocks.html          dependencies.html               plugin-management.html         surefire-report.html
config_coding.html          dependency-info.html            plugin-updates-report.html     taglist.html
config_design.html          dependency-updates-report.html  plugins.html                   team.html
config_filefilters.html     distribution-management.html    pmd.html                       version
config_filters.html         dsm                             project-info.html              writingchecks.html
config_header.html          eclipse.html                    project-reports.html           writingfilefilters.html
config_imports.html         extending.html                  property_types.html            writingfilters.html
config_javadoc.html         failsafe-report.html            releasenotes.html              writingjavadocchecks.html
config_metrics.html         files                           releasenotes_old_1-0_5-9.html  writinglisteners.html
config_misc.html            google_style.html               releasenotes_old_6-0_7-8.html  xref
config_modifier.html        idea.html                       report_issue.html              xref-test

```

Reply:
Each page has header `| Version: 10.4`
Can we reliable parse it out ? For example from index.html

Reply:
I read that parsing HTML with regex is generally a bad idea.

We can still do something like that if we are certain that the latest tag is already in the new release version.
```
20:21 $ git tag --sort=creatordate | tail -2 | head -1
checkstyle-10.3.4

```

Reply:
Parsing is bad idea, but grep on short content was always good enough.

Reply:
Done.
https://github.com/stoyanK7/checkstyle/actions/runs/3539713984/jobs/5941937129#step:4:21
```
Extracting previous release version
PREVIOUS RELEASE version:10.4
```